compiler/msgs.nim

```diff
@@ -734,20 +734,19 @@ proc outWriteln*(s: string) =
     flushFile(stdout)
 
 proc msgWriteln*(s: string) =
-  ## Writes to stdout. If --stdout option is given, writes to stderr instead.
+  ## Writes to stderr. If --stdout option is given, writes to stdout instead.
 
   #if gCmd == cmdIdeTools and optCDebug notin gGlobalOptions: return
 
   if not isNil(writelnHook):
     writelnHook(s)
   elif optStdout in gGlobalOptions:
-    if eStdErr in errorOutputs:
-      writeLine(stderr, s)
-      flushFile(stderr)
-  else:
     if eStdOut in errorOutputs:
       writeLine(stdout, s)
       flushFile(stdout)
+  else:
+    if eStdErr in errorOutputs:
+      writeLine(stderr, s)
```

Comment:
Quoting ISO/IEC 9899:TC3 C99 std:

> At program startup, three text streams are predefined and need not be opened explicitly — standard input (for reading conventional input), standard output (for writing conventional output), and standard error (for writing diagnostic output). As initially opened, **the standard error stream is not fully buffered**; the standard input and standard output streams are fully buffered if and only if the stream can be determined not to refer to an interactive device.

This essentially means that `stderr` is either line-buffered or unbuffered. Therefore we don't need flush after new line, because newline itself implicitly flushes.

Anyways just to be sure I'll do some test on Windows just to see if Microsoft is not violating that.